howtos/how-to-set-an-away-message/server.js

```diff
@@ -0,0 +1,185 @@
+// Whenever you start the server, we make sure the two example users are created
+// recreate the two conversations, and send messages from the example users
+
+async function setupConversation(i) {
+  const conversationId = `autoReplyExample${i}`;
+  const userId = `autoReplyExampleUser${i}`;
+
+  // Delete the conversation (if it exists)
+  await fetch(`${basePath}/v1/${appId}/conversations/${conversationId}`, {
+    method: "delete",
+    headers: {
+      Authorization: `Bearer ${secretKey}`,
+    },
+  });
+
+  // Create a new conversation
+  await fetch(`${basePath}/v1/${appId}/conversations/${conversationId}`, {
+    method: "put",
+    headers: {
+      "Content-Type": "application/json",
+      Authorization: `Bearer ${secretKey}`,
+    },
+    body: JSON.stringify({
+      participants: ["autoReplyExampleSupportAgent", userId],
+    }),
+  });
+
+  // Send a message from the user to make sure it will show up in the conversation list
+  await fetch(
+    `${basePath}/v1/${appId}/conversations/${conversationId}/messages`,
+    {
+      method: "post",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${secretKey}`,
+      },
+      body: JSON.stringify([
+        {
+          text: "Everything is broken again!",
+          sender: userId,
+          type: "UserMessage",
+        },
+      ]),
+    }
+  );
+}
+
+async function setup() {
+  const supportAgent = fetch(
+    `${basePath}/v1/${appId}/users/autoReplyExampleSupportAgent`,
+    {
+      method: "put",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${secretKey}`,
+      },
+      body: JSON.stringify({
+        name: "Alice",
+        email: ["alice@example.com"],
+        photoUrl: "https://talkjs.com/images/avatar-1.jpg",
+        role: "support",
+        welcomeMessage: "Hey there! How can I help?",
+      }),
+    }
+  );
+
+  const user1 = fetch(`${basePath}/v1/${appId}/users/autoReplyExampleUser1`, {
+    method: "put",
+    headers: {
+      "Content-Type": "application/json",
+      Authorization: `Bearer ${secretKey}`,
+    },
+    body: JSON.stringify({
+      name: "Sebastian",
+      email: ["Sebastian@example.com"],
+      photoUrl: "https://talkjs.com/images/avatar-5.jpg",
+      role: "customer",
+      welcomeMessage: "Hi!",
+    }),
+  });
+
+  const user2 = fetch(`${basePath}/v1/${appId}/users/autoReplyExampleUser2`, {
+    method: "put",
+    headers: {
+      "Content-Type": "application/json",
+      Authorization: `Bearer ${secretKey}`,
+    },
+    body: JSON.stringify({
+      name: "Bob",
+      email: ["Bob@example.com"],
+      photoUrl: "https://talkjs.com/images/avatar-4.jpg",
+      role: "customer",
+      welcomeMessage: "Hello!",
+    }),
+  });
+
+  await supportAgent;
+  await user1;
+  await user2;
+
+  const conv1 = setupConversation(1);
+  const conv2 = setupConversation(2);
+
+  await conv1;
+  await conv2;
+}
+
+setup();
```

Comment:
We probably don't actually need to create the conversations, only the users. The whole idea is that it's an auto-reply in case a user messages you, so the conversation wouldn't already need to exist

Reply:
It does work quite well as a demonstration though, having the conversations created and then the auto-reply in response if it's out of hours